SecurityException in WifiManager

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:2750)
  at android.app.ActivityThread.access$1800 (ActivityThread.java:157)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1433)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:148)
  at android.app.ActivityThread.main (ActivityThread.java:5551)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:730)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:620)
Caused by: java.lang.SecurityException: 
  at android.os.Parcel.readException (Parcel.java:1599)
  at android.os.Parcel.readException (Parcel.java:1552)
  at android.net.wifi.IWifiManager$Stub$Proxy.getScanResults (IWifiManager.java:1150)
  at android.net.wifi.WifiManager.getScanResults (WifiManager.java:1365)
  at de.tum.in.tumcampusapp.services.ScanResultsAvailableReceiver.onReceive (ScanResultsAvailableReceiver.java:73)
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:2743)
```

--- a/app/src/main/java/de/tum/in/tumcampusapp/services/ScanResultsAvailableReceiver.java
+++ b/app/src/main/java/de/tum/in/tumcampusapp/services/ScanResultsAvailableReceiver.java
@@ -15,7 +15,6 @@ import android.net.wifi.WifiManager;
 import android.os.Build;
 import android.support.v4.app.NotificationCompat;
 import android.support.v4.content.ContextCompat;
-import android.util.Log;
 
 import java.util.Calendar;
 import java.util.List;

--- a/app/src/main/java/de/tum/in/tumcampusapp/services/ScanResultsAvailableReceiver.java
+++ b/app/src/main/java/de/tum/in/tumcampusapp/services/ScanResultsAvailableReceiver.java
@@ -15,6 +15,7 @@ import android.net.wifi.WifiManager;
 import android.os.Build;
 import android.support.v4.app.NotificationCompat;
 import android.support.v4.content.ContextCompat;
+import android.util.Log;
 
 import java.util.Calendar;
 import java.util.List;
@@ -69,6 +70,12 @@ public class ScanResultsAvailableReceiver extends BroadcastReceiver {
         boolean locationsEnabled = ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED;
         boolean wifiScansEnabled = Utils.getInternalSettingBool(context, WifiMeasurementManager.WIFI_SCANS_ALLOWED, false);
         boolean nextScanScheduled = false;
+
+        if (!locationsEnabled) {
+            //Stop here as wifi.getScanResults will either return an empty list or throw an exception (on android 6.0.0)
+            return;
+        }
+
         WifiScanHandler wifiScanHandler = WifiScanHandler.getInstance(context);
         List<ScanResult> scan = wifi.getScanResults();
         for (final ScanResult network : scan) {
@@ -82,7 +89,7 @@ public class ScanResultsAvailableReceiver extends BroadcastReceiver {
             }
 
             //if user allowed us to store his signal strength, store measurement to the local DB and later sync to remote
-            if (locationsEnabled && wifiScansEnabled) {
+            if (wifiScansEnabled) {
                 storeWifiMeasurement(context, network);
                 nextScanScheduled = true;
             }

